src/main/java/uk/ac/ebi/tsc/portal/api/team/controller/TeamRestController.java

```diff
@@ -683,6 +683,49 @@ public TeamResource getTeamByName(HttpServletRequest request, Principal principa
 		return new ResponseEntity<>("User  request was successfully sent to team owner " + teamResource.getName(), HttpStatus.OK);
 	}
 
+	@RequestMapping(value = "/{teamName:.+}/contactemail/", method = RequestMethod.PUT)
+	public ResponseEntity<?> addTeamContactEmails(HttpServletRequest request, Principal principal,
+												  @PathVariable("teamName") String teamName,
+												  @RequestBody String emails) {
+		logger.info("User " + principal.getName() + " requested adding contacts to team " + teamName);
+
+		if(teamName == null || teamName.isEmpty()){
+			throw new TeamNameInvalidInputException("Team name should not be empty");
+		}
+
+		logger.info("Checking if user is team owner");
+		Team team = teamService.findByName(teamName);
+		if(!team.getAccount().getUsername().equals(principal.getName())){
+			throw new TeamNotFoundException(team.getName() + " is not found/not accessible by user");
```

Comment:
This error message is misleading and the error type is rather wrong. What does it mean that "Team is not found/accesible"?
It is a user without permissions trying to perform an operation, right? Why not 403?

Reply:
We can do it 2 ways: 
1. Acknowledge team is present, but throw 403 or
2.  just not informing the user that such a team exists, by throwing TeamNotFoundException.(this is what I have followed)

We can discuss which response would be better

Reply:
Anyway the user will know the presence of a team. Forbidden makes more sense, so have implemented what you suggested